query/src/main/java/org/infinispan/query/impl/BaseEntityLoader.java

```diff
@@ -0,0 +1,82 @@
+package org.infinispan.query.impl;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+import org.hibernate.search.engine.common.timing.Deadline;
+import org.hibernate.search.mapper.pojo.loading.spi.PojoSelectionEntityLoader;
+import org.infinispan.AdvancedCache;
+import org.infinispan.query.core.stats.impl.LocalQueryStatistics;
+
+abstract class BaseEntityLoader<E, T> implements PojoSelectionEntityLoader<EntityLoaded<E>> {
+
+   protected final AdvancedCache<Object, Object> cache;
+   private final LocalQueryStatistics queryStatistics;
+
+   BaseEntityLoader(AdvancedCache<Object, Object> cache, LocalQueryStatistics queryStatistics) {
+      this.cache = Objects.requireNonNull(cache);
+      this.queryStatistics = Objects.requireNonNull(queryStatistics);
+   }
+
+   @Override
+   public final List<EntityLoaded<E>> loadBlocking(List<?> identifiers, Deadline deadline) {
+      assert identifiers != null;
+      if (identifiers.isEmpty()) {
+         return List.of();
+      }
+
+      long start = queryStatistics.isEnabled() ? System.nanoTime() : 0;
```

Comment:
Shouldn't we be using the TimeService for these ?

Reply:
Yes. The query module is still using `System.nanoTime()` everywhere and we haven't updated it yet. I can update it here!